assets/js/googlesitekit/data/create-notifications-store.js

```diff
@@ -0,0 +1,239 @@
+ *
+ * The three required parameters hook up the store to the respective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @private
+ * @param {string} type       The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier The data identifier, eg. a module slug like 'search-console'.
+ * @param {string} datapoint  The endpoint to request data from, e.g. 'notifications'.
+ * @return {Object} The notifications store object.
+ */
+export const createNotificationsStore = ( type, identifier, datapoint ) => {
+	const INITIAL_STATE = {
+		serverNotifications: undefined,
+		clientNotifications: {},
+		isFetchingNotifications: false,
+	};
+
+	const actions = {
+		/**
+		 * Adds a notification to the store.
+		 *
+		 * @since n.e.x.t
+		 * @param {Object} notification Notification object to add.
+		 * @return {Object} Redux-style action.
+		 */
+		addNotification( notification ) {
+			return {
+				payload: { notification },
+				type: ADD_NOTIFICATION,
+			};
+		},
+
+		/**
+		 * Removes a notification from the store.
+		 *
+		 * @since n.e.x.t
+		 * @param {string} id ID of the notification to remove.
+		 * @return {Object} Redux-style action.
+		 */
+		removeNotification( id ) {
+			return {
+				payload: { id },
+				type: REMOVE_NOTIFICATION,
+			};
+		},
+
+		/**
+		 * Dispatches an action that creates an HTTP request to the notifications endpoint.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 * @return {Object} Redux-style action.
+		 */
+		fetchNotifications() {
+			return {
+				payload: {},
+				type: FETCH_NOTIFICATIONS,
+			};
+		},
+
+		/**
+		 * Stores notifications received from the REST API.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 * @param {Array} notifications Notifications from the API.
+		 * @return {Object} Redux-style action.
+		 */
+		receiveNotifications( notifications ) {
+			invariant( notifications, 'notifications is required.' );
+
+			return {
+				payload: { notifications },
+				type: RECEIVE_NOTIFICATIONS,
+			};
+		},
+
+		/**
+		 * Dispatches an action signifying the `fetchNotifications` side-effect failed.
+		 *
+		 * @since n.e.x.t
+		 * @private
+		 * @return {Object} Redux-style action.
+		 */
+		receiveNotificationsFailed() {
+			return {
+				payload: {},
+				type: RECEIVE_NOTIFICATIONS_FAILED,
+			};
+		},
+	};
+
+	const controls = {
+		[ FETCH_NOTIFICATIONS ]: () => {
+			return get( type, identifier, datapoint );
+		},
+	};
+
+	const reducer = ( state = INITIAL_STATE, action ) => {
+		switch ( action.type ) {
+			case ADD_NOTIFICATION: {
+				const { notification } = action.payload;
+
+				return {
+					...state,
+					clientNotifications: {
+						...state.clientNotifications,
+						[ notification.id ]: notification,
+					},
+				};
+			}
+
+			case REMOVE_NOTIFICATION: {
+				const { id } = action.payload;
+
+				const newNotifications = { ...state.clientNotifications };
+				delete newNotifications[ id ];
+
+				return {
```

Comment:
See above re: my thoughts on this being an unexpected API—client/server differentiation of notifications is not made to consumers of this store, but the store silently treats them differently on remove.